libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs

```diff
@@ -303,29 +317,69 @@ private static void AddProperty(JObject obj, string key, JToken value)
 
-        private Task<RecognizerResult> RecognizeInternalAsync(string utterance, CancellationToken ct)
+        private static void AddProperties(LuisResult luis, RecognizerResult result)
         {
-            if (string.IsNullOrEmpty(utterance))
-            {
-                throw new ArgumentNullException(nameof(utterance));
-            }
-
-            var luisRequest = new LuisRequest(utterance);
-            _luisOptions.Apply(luisRequest);
-            return RecognizeAsync(luisRequest, ct, _luisRecognizerOptions.Verbose);
+            if (luis.SentimentAnalysis != null)
+            {
+                result.Properties.Add("sentiment", new JObject(
+                    new JProperty("label", luis.SentimentAnalysis.Label),
+                    new JProperty("score", luis.SentimentAnalysis.Score)));
+            }
         }
 
-        private async Task<RecognizerResult> RecognizeAsync(LuisRequest request, CancellationToken ct, bool verbose)
+        private async Task<RecognizerResult> RecognizeInternalAsync(ITurnContext context, CancellationToken ct)
         {
-            var luisResult = await _luisService.QueryAsync(request, ct).ConfigureAwait(false);
+            BotAssert.ContextNotNull(context);
+
+            if (context.Activity.Type != ActivityTypes.Message)
+            {
+                return null;
+            }
+
+            var utterance = context.Activity?.AsMessageActivity()?.Text;
+
+            if (string.IsNullOrWhiteSpace(utterance))
+            {
+                throw new ArgumentNullException(nameof(utterance));
+            }
+
+            var luisResult = await runtime.Prediction.ResolveAsync(
+                application.ApplicationId,
+                utterance,
+                timezoneOffset: options.TimezoneOffset,
+                verbose: options.Verbose,
+                staging: options.Staging,
+                spellCheck: options.SpellCheck,
+                bingSpellCheckSubscriptionKey: options.BingSpellCheckSubscriptionKey,
+                log: options.Log,
+                cancellationToken: ct).ConfigureAwait(false);
+
             var recognizerResult = new RecognizerResult
             {
-                Text = request.Query,
+                Text = utterance,
                 AlteredText = luisResult.AlteredQuery,
                 Intents = GetIntents(luisResult),
-                Entities = ExtractEntitiesAndMetadata(luisResult.Entities, luisResult.CompositeEntities, verbose),
+                Entities = ExtractEntitiesAndMetadata(luisResult.Entities, luisResult.CompositeEntities, options.IncludeInstanceData ?? true),
             };
-            recognizerResult.Properties.Add("luisResult", luisResult);
+            AddProperties(luisResult, recognizerResult);
+            if (includeApiResults)
+            {
+                recognizerResult.Properties.Add("luisResult", luisResult);
+            }
+
+            var traceInfo = JObject.FromObject(
+                new
```

Comment:
Consider creating a strong type here. Using an anonymous type here seems problematic over the long term.

Reply:
This was the lowest impact change to make.  It used to be strongly typed, but the strong types needed to change.  If they did then they would also have to change in the emulator and the V3 SDK.  Instead we opted for keeping the format and copying it.  This also prevented exposing what are really internal classes.